doc/developer/platform/ux.md

```diff
@@ -1,20 +1,945 @@
+# Materialize Platform: User Experience
 
-⚠️ **WARNING!** ⚠️ This doc is a work in progress!
+## How to read this document
 
-The intended user experience drives some of the concepts that we'll want to reflect in our code.
-The intended user experience in turn reflects the practical realities of the tools we have access to.
+⚠️ **WARNING!** ⚠️ This document is a work in progress! While the underlying
+architecture is firm, the particular mapping from architecture to user-facing
+concepts described in this document is still up for debate.
 
-The coarsest level of granularity of Materialize Platform is the ACCOUNT.
-Within an ACCOUNT there may be multiple REGIONs, which correspond roughly to availability zones.
-A REGION is the unit within which it is reasonable to transfer data, and across which that expectation doesn't exist.
+This document is a proposal for how we expose the
+[architecture](architecture-db.md) of Materialize Platform to users. The
+intended user experience drives some of the concepts that we'll want to reflect
+in our code. The intended user experience in turn reflects the practical
+realities of the tools we have access to.
 
-Within each REGION there may be multiple CLUSTERs, which correspond roughly to timely dataflow instances.
-A CLUSTER contains indexes and is the unit within which it is possible to share indexes.
+Terms in this document that represent key user-facing concepts are introduced
+with *italics*. The underlying components may use different terminology
+internally, but any user-facing interfaces should use the term specified in this
+document. For example, the COMPUTE layer is free to use the term COMPUTE
+INSTANCE internally, but the user-facing documentation, web UI, and SQL API
+should all consistently use the term *cluster*.
 
-Within each REGION there may be multiple TIMELINEs, which correspond roughly to coordinator timelines.
-A TIMELINE contains collections, and is the unit within which it is possible to query across multiple collections
+## Overview
 
-The main restrictions we plan to impose on users is that their access to data is always associated with a REGION, CLUSTER, and TIMELINE, and attempts to access data across these may result in poorer performance, or explicit query rejection.
+Depicted below are the tangible concepts that Materialize Platform exposes to
+users.
 
-There is some work to do to present these concepts to users, but prior work exists with e.g. SQL `database`s for TIMELINE and Snowflake's `USE WAREHOUSE` command to pilot CLUSTER.
+![Diagram of user-facing concepts](./assets/user-concepts.png)
+
+An [*account*](#account) represents a customer of Materialize Platform. An
+account has [*users*](#user) and manages [*billing*](#billing).
+
+A [*region*](#region) is a specific region of a specific cloud provider (e.g.,
+aws:us-east-1). The region boundary is strong: besides users and billing,
+nothing is shared between regions in the same account.
```

Comment:
Are users able to do _anything_ across regions?  Say, for data locality compliance reasons, a user partitions raw data across regions -- (how) would they be able to use materialize to analyze their global data?  Even if we don't plan to support it "soon", I think there's an argument to be made to do this eventually

Reply:
> Are users able to do _anything_ across regions? Say, for data locality compliance reasons, a user partitions raw data across regions -- (how) would they be able to use materialize to analyze their global data? Even if we don't plan to support it "soon", I think there's an argument to be made to do this eventually

I don't know how satisfying you'll find this answer, but they're welcome to ingest all their data into a single Materialize region. Nothing stops a customer from having their EU region of Materialize Platform read from a Kafka broker in the US, say. They'll just have to eat the network transfer costs.